clients/src/main/java/org/apache/kafka/common/errors/PendingOffsetException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
```

Comment:
If this is not supposed to be thrown out of the public APIs anyways I'd suggest moving to `org.apache.kafka.clients.consumer.internals`. In a hind-sight we should do that for some of the existing ones too but since they are already out in the wild we cannot take them back, but at least we should stop leaking them out if it is not necessary.

Reply:
The difficulty for doing that is `GroupCoordinator.scala` and `Errors.java` don't have access to the client internal package, so that we couldn't return it correctly from server side.

Reply:
How about moving it to `org.apache.kafka.clients.consumer` then? It already has a non-public `RetriableCommitFailed` and a public `CommitFailed` --- it is a nitpick and there's really not strict rules, but I feel more comfortable having it there than `o.a.k.common.errors`.

Reply:
EDIT: seems that for any broker returned error code they have to be in this package, so ignore my last comment.